{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "d919e989",
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "df332407",
   "metadata": {},
   "outputs": [],
   "source": [
    "today_date = datetime.now().strftime(\"%m.%d.%Y\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "fe746941",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:root:Running STAR alignment for SRR12801019 with command: /private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR --runThreadN 10 --genomeDir /private/groups/brookslab/reference_indices/star_hg38_rausharm --readFilesIn /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801019_pass_1.fastq /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801019_pass_2.fastq --outFileNamePrefix /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801019/SRR12801019_ --outSAMtype BAM SortedByCoordinate\n",
      "\t/private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR --runThreadN 10 --genomeDir /private/groups/brookslab/reference_indices/star_hg38_rausharm --readFilesIn /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801019_pass_1.fastq /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801019_pass_2.fastq --outFileNamePrefix /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801019/SRR12801019_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 13 17:13:00 ..... started STAR run\n",
      "Mar 13 17:13:01 ..... loading genome\n",
      "Mar 13 17:13:29 ..... started mapping\n",
      "Mar 13 17:19:49 ..... finished mapping\n",
      "Mar 13 17:19:57 ..... started sorting BAM\n",
      "Mar 13 17:22:22 ..... finished successfully\n",
      "INFO:root:BAM manifest updated with new entries.\n",
      "INFO:root:Alignment completed.\n"
     ]
    }
   ],
   "source": [
    "!python3 alignment.py --star-path \"/private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR\" \\\n",
    "--genomeDir \"/private/groups/brookslab/reference_indices/star_hg38_rausharm\" \\\n",
    "--readFilesIn \"/private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801019_pass_1.fastq\" \"/private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801019_pass_2.fastq\" \\\n",
    "--output-dir \"/private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/\" \\\n",
    "--metadata \"SF3B1\" \\\n",
    "--condition \"Reference\" \\\n",
    "--paired"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8cdac442",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:root:Running STAR alignment for SRR12801023 with command: /private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR --runThreadN 10 --genomeDir /private/groups/brookslab/reference_indices/star_hg38_rausharm --readFilesIn /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801023_pass_1.fastq /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801023_pass_2.fastq --outFileNamePrefix /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801023/SRR12801023_ --outSAMtype BAM SortedByCoordinate\n",
      "\t/private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR --runThreadN 10 --genomeDir /private/groups/brookslab/reference_indices/star_hg38_rausharm --readFilesIn /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801023_pass_1.fastq /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801023_pass_2.fastq --outFileNamePrefix /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801023/SRR12801023_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 13 17:22:27 ..... started STAR run\n",
      "Mar 13 17:22:28 ..... loading genome\n",
      "Mar 13 17:22:56 ..... started mapping\n",
      "Mar 13 17:32:35 ..... finished mapping\n",
      "Mar 13 17:32:41 ..... started sorting BAM\n",
      "Mar 13 17:34:59 ..... finished successfully\n",
      "INFO:root:BAM manifest updated with new entries.\n",
      "INFO:root:Alignment completed.\n"
     ]
    }
   ],
   "source": [
    "!python3 alignment.py --star-path \"/private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR\" \\\n",
    "--genomeDir \"/private/groups/brookslab/reference_indices/star_hg38_rausharm\" \\\n",
    "--readFilesIn \"/private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801023_pass_1.fastq\" \"/private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801023_pass_2.fastq\" \\\n",
    "--output-dir \"/private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/\" \\\n",
    "--metadata \"SF3B1\" \\\n",
    "--condition \"Reference\" \\\n",
    "--paired"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "393e4015",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:root:Running STAR alignment for SRR12801020 with command: /private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR --runThreadN 10 --genomeDir /private/groups/brookslab/reference_indices/star_hg38_rausharm --readFilesIn /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801020_pass_1.fastq /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801020_pass_2.fastq --outFileNamePrefix /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801020/SRR12801020_ --outSAMtype BAM SortedByCoordinate\n",
      "\t/private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR --runThreadN 10 --genomeDir /private/groups/brookslab/reference_indices/star_hg38_rausharm --readFilesIn /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801020_pass_1.fastq /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801020_pass_2.fastq --outFileNamePrefix /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801020/SRR12801020_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 13 17:35:06 ..... started STAR run\n",
      "Mar 13 17:35:07 ..... loading genome\n",
      "Mar 13 17:36:12 ..... started mapping\n",
      "Mar 13 17:55:35 ..... finished mapping\n",
      "Mar 13 17:55:43 ..... started sorting BAM\n",
      "Mar 13 17:58:40 ..... finished successfully\n",
      "INFO:root:BAM manifest updated with new entries.\n",
      "INFO:root:Alignment completed.\n"
     ]
    }
   ],
   "source": [
    "!python3 alignment.py --star-path \"/private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR\" \\\n",
    "--genomeDir \"/private/groups/brookslab/reference_indices/star_hg38_rausharm\" \\\n",
    "--readFilesIn \"/private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801020_pass_1.fastq\" \"/private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801020_pass_2.fastq\" \\\n",
    "--output-dir \"/private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/\" \\\n",
    "--metadata \"SUGP1\" \\\n",
    "--condition \"Sample\" \\\n",
    "--paired"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "fae146f2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:root:Running STAR alignment for SRR12801024 with command: /private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR --runThreadN 10 --genomeDir /private/groups/brookslab/reference_indices/star_hg38_rausharm --readFilesIn /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801024_pass_1.fastq /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801024_pass_2.fastq --outFileNamePrefix /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801024/SRR12801024_ --outSAMtype BAM SortedByCoordinate\n",
      "\t/private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR --runThreadN 10 --genomeDir /private/groups/brookslab/reference_indices/star_hg38_rausharm --readFilesIn /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801024_pass_1.fastq /private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801024_pass_2.fastq --outFileNamePrefix /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801024/SRR12801024_ --outSAMtype BAM SortedByCoordinate\n",
      "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
      "Mar 13 17:58:49 ..... started STAR run\n",
      "Mar 13 17:58:49 ..... loading genome\n",
      "Mar 13 17:59:16 ..... started mapping\n",
      "Mar 13 18:16:55 ..... finished successfully\n",
      "INFO:root:BAM manifest updated with new entries.\n",
      "INFO:root:Alignment completed.\n"
     ]
    }
   ],
   "source": [
    "!python3 alignment.py --star-path \"/private/groups/brookslab/bin/STAR-2.7.10a/bin/Linux_x86_64/STAR\" \\\n",
    "--genomeDir \"/private/groups/brookslab/reference_indices/star_hg38_rausharm\" \\\n",
    "--readFilesIn \"/private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801024_pass_1.fastq\" \"/private/groups/brookslab/scratch/rausharm/sra_acc.02.09.24/fastq/SRR12801024_pass_2.fastq\" \\\n",
    "--output-dir \"/private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/\" \\\n",
    "--metadata \"SUGP1\" \\\n",
    "--condition \"Sample\" \\\n",
    "--paired"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "fcedaf3e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:root:Running samtools flagstat on /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801019/SRR12801019_Aligned.sortedByCoord.out.bam\n",
      "INFO:root:Running samtools stats on /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801019/SRR12801019_Aligned.sortedByCoord.out.bam\n",
      "INFO:root:Running samtools flagstat on /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801020/SRR12801020_Aligned.sortedByCoord.out.bam\n",
      "INFO:root:Running samtools stats on /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801020/SRR12801020_Aligned.sortedByCoord.out.bam\n",
      "INFO:root:Running samtools flagstat on /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801023/SRR12801023_Aligned.sortedByCoord.out.bam\n",
      "INFO:root:Running samtools stats on /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801023/SRR12801023_Aligned.sortedByCoord.out.bam\n",
      "INFO:root:Running samtools flagstat on /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801024/SRR12801024_Aligned.sortedByCoord.out.bam\n",
      "INFO:root:Running samtools stats on /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801024/SRR12801024_Aligned.sortedByCoord.out.bam\n"
     ]
    }
   ],
   "source": [
    "!python3 bam_qc_samtools.py \"/private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ddecb9c8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finding junctions from 4 BAM files...\n",
      "bam: /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801019/SRR12801019_Aligned.sortedByCoord.out.bam\n",
      "number of junctions found: 517327\n",
      "saved to bed: /private/groups/brookslab/rausharm/RBPSig/03.11.24_junction_beds/SRR12801019_Aligned.sortedByCoord.out.junc.bed\n",
      "bam: /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801023/SRR12801023_Aligned.sortedByCoord.out.bam\n",
      "number of junctions found: 523173\n",
      "saved to bed: /private/groups/brookslab/rausharm/RBPSig/03.11.24_junction_beds/SRR12801023_Aligned.sortedByCoord.out.junc.bed\n",
      "bam: /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801020/SRR12801020_Aligned.sortedByCoord.out.bam\n",
      "number of junctions found: 576971\n",
      "saved to bed: /private/groups/brookslab/rausharm/RBPSig/03.11.24_junction_beds/SRR12801020_Aligned.sortedByCoord.out.junc.bed\n",
      "bam: /private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/SRR12801024/SRR12801024_Aligned.sortedByCoord.out.bam\n",
      "number of junctions found: 576484\n",
      "saved to bed: /private/groups/brookslab/rausharm/RBPSig/03.11.24_junction_beds/SRR12801024_Aligned.sortedByCoord.out.junc.bed\n",
      "new manifest written to: /private/groups/brookslab/rausharm/RBPSig/03.11.24_manifest.txt\n",
      "Command executed successfully. Output files are prefixed with: /private/groups/brookslab/rausharm/RBPSig/03.11.24\n",
      "Parsing manifest...\n",
      "\tDone [0:00:0.00]\n",
      "Getting all junctions from 4 files...\n",
      "\tDone [0:00:4.85]\n",
      "Finding clusters from 314403 junctions...\n",
      "\tDone [0:00:1.73]\n",
      "Writing cluster file...\n",
      "\tDone [0:00:0.69]\n",
      "Writing junction bed file...\n",
      "\tDone [0:00:0.84]\n",
      "Gathering junction counts...\n",
      "\tDone [0:00:3.57]\n",
      "Writing inclusion counts...\n",
      "\tDone [0:00:1.46]\n",
      "Calculating PS values...\n",
      "/private/home/rausharm/mesa/mesa/MESA.py:302: RuntimeWarning: invalid value encountered in divide\n",
      "  psi[self.junctionIndex[junction],:] = inclusions / (inclusions + exclusions)\n",
      "\tDone [0:00:1.80]\n",
      "Writing PS values...\n",
      "\tDone [0:00:1.49]\n",
      "All done [0:00:16.45]\n",
      "mesa quant command executed successfully. Output files are prefixed with: /private/groups/brookslab/rausharm/RBPSig/03.11.24\n",
      "Counts loaded from /private/groups/brookslab/rausharm/RBPSig/03.11.24_inclusionCounts.tsv ...\n",
      "Clusters loaded from /private/groups/brookslab/rausharm/RBPSig/03.11.24_allClusters.tsv ...\n",
      "Analyzing pairs:\n",
      "SRR12801019_SRR12801023,SRR12801019_SRR12801020,SRR12801019_SRR12801024,SRR12801023_SRR12801020,SRR12801023_SRR12801024,SRR12801020_SRR12801024\n",
      "[0 / 314403] events analyzed...\n",
      "[50 / 314403] events analyzed...\n",
      "[100 / 314403] events analyzed...\n",
      "[150 / 314403] events analyzed...\n",
      "[200 / 314403] events analyzed...\n",
      "[250 / 314403] events analyzed...\n",
      "[300 / 314403] events analyzed...\n",
      "[350 / 314403] events analyzed...\n",
      "[400 / 314403] events analyzed...\n",
      "[450 / 314403] events analyzed...\n",
      "[500 / 314403] events analyzed...\n",
      "[550 / 314403] events analyzed...\n",
      "[600 / 314403] events analyzed...\n",
      "[650 / 314403] events analyzed...\n",
      "[700 / 314403] events analyzed...\n",
      "[750 / 314403] events analyzed...\n",
      "[800 / 314403] events analyzed...\n",
      "[850 / 314403] events analyzed...\n",
      "[900 / 314403] events analyzed...\n",
      "[950 / 314403] events analyzed...\n",
      "[1000 / 314403] events analyzed...\n",
      "[1050 / 314403] events analyzed...\n",
      "[1100 / 314403] events analyzed...\n",
      "[1150 / 314403] events analyzed...\n",
      "[1200 / 314403] events analyzed...\n",
      "[1250 / 314403] events analyzed...\n",
      "[1300 / 314403] events analyzed...\n",
      "[1350 / 314403] events analyzed...\n",
      "[1400 / 314403] events analyzed...\n",
      "[1450 / 314403] events analyzed...\n",
      "[1500 / 314403] events analyzed...\n",
      "[1550 / 314403] events analyzed...\n",
      "[1600 / 314403] events analyzed...\n",
      "[1650 / 314403] events analyzed...\n",
      "[1700 / 314403] events analyzed...\n",
      "[1750 / 314403] events analyzed...\n",
      "[1800 / 314403] events analyzed...\n",
      "[1850 / 314403] events analyzed...\n",
      "[1900 / 314403] events analyzed...\n",
      "[1950 / 314403] events analyzed...\n",
      "[2000 / 314403] events analyzed...\n",
      "[2050 / 314403] events analyzed...\n",
      "[2100 / 314403] events analyzed...\n",
      "[2150 / 314403] events analyzed...\n",
      "[2200 / 314403] events analyzed...\n",
      "[2250 / 314403] events analyzed...\n",
      "[2300 / 314403] events analyzed...\n",
      "[2350 / 314403] events analyzed...\n",
      "[2400 / 314403] events analyzed...\n",
      "[2450 / 314403] events analyzed...\n",
      "[2500 / 314403] events analyzed...\n",
      "[2550 / 314403] events analyzed...\n",
      "[2600 / 314403] events analyzed...\n",
      "[2650 / 314403] events analyzed...\n",
      "[2700 / 314403] events analyzed...\n",
      "[2750 / 314403] events analyzed...\n",
      "[2800 / 314403] events analyzed...\n",
      "[2850 / 314403] events analyzed...\n",
      "[2900 / 314403] events analyzed...\n",
      "[2950 / 314403] events analyzed...\n",
      "[3000 / 314403] events analyzed...\n",
      "[3050 / 314403] events analyzed...\n",
      "[3100 / 314403] events analyzed...\n",
      "[3150 / 314403] events analyzed...\n",
      "[3200 / 314403] events analyzed...\n",
      "[3250 / 314403] events analyzed...\n",
      "[3300 / 314403] events analyzed...\n",
      "[3350 / 314403] events analyzed...\n",
      "[3400 / 314403] events analyzed...\n",
      "[3450 / 314403] events analyzed...\n",
      "[3500 / 314403] events analyzed...\n",
      "[3550 / 314403] events analyzed...\n",
      "[3600 / 314403] events analyzed...\n",
      "[3650 / 314403] events analyzed...\n",
      "[3700 / 314403] events analyzed...\n",
      "[3750 / 314403] events analyzed...\n",
      "[3800 / 314403] events analyzed...\n",
      "[3850 / 314403] events analyzed...\n",
      "[3900 / 314403] events analyzed...\n",
      "[3950 / 314403] events analyzed...\n",
      "[4000 / 314403] events analyzed...\n",
      "[4050 / 314403] events analyzed...\n",
      "[4100 / 314403] events analyzed...\n",
      "[4150 / 314403] events analyzed...\n",
      "[4200 / 314403] events analyzed...\n",
      "[4250 / 314403] events analyzed...\n",
      "[4300 / 314403] events analyzed...\n",
      "[4350 / 314403] events analyzed...\n",
      "[4400 / 314403] events analyzed...\n",
      "[4450 / 314403] events analyzed...\n",
      "[4500 / 314403] events analyzed...\n",
      "[4550 / 314403] events analyzed...\n",
      "[4600 / 314403] events analyzed...\n",
      "[4650 / 314403] events analyzed...\n",
      "[4700 / 314403] events analyzed...\n",
      "[4750 / 314403] events analyzed...\n",
      "[4800 / 314403] events analyzed...\n",
      "[4850 / 314403] events analyzed...\n",
      "[4900 / 314403] events analyzed...\n",
      "[4950 / 314403] events analyzed...\n",
      "[5000 / 314403] events analyzed...\n",
      "[5050 / 314403] events analyzed...\n",
      "[5100 / 314403] events analyzed...\n",
      "[5150 / 314403] events analyzed...\n",
      "[5200 / 314403] events analyzed...\n",
      "[5250 / 314403] events analyzed...\n",
      "[5300 / 314403] events analyzed...\n",
      "[5350 / 314403] events analyzed...\n",
      "[5400 / 314403] events analyzed...\n",
      "[5450 / 314403] events analyzed...\n",
      "[5500 / 314403] events analyzed...\n",
      "[5550 / 314403] events analyzed...\n",
      "[5600 / 314403] events analyzed...\n",
      "[5650 / 314403] events analyzed...\n",
      "[5700 / 314403] events analyzed...\n",
      "[5750 / 314403] events analyzed...\n",
      "[5800 / 314403] events analyzed...\n",
      "[5850 / 314403] events analyzed...\n",
      "[5900 / 314403] events analyzed...\n",
      "[5950 / 314403] events analyzed...\n",
      "[6000 / 314403] events analyzed...\n",
      "[6050 / 314403] events analyzed...\n",
      "[6100 / 314403] events analyzed...\n",
      "[6150 / 314403] events analyzed...\n",
      "[6200 / 314403] events analyzed...\n",
      "[6250 / 314403] events analyzed...\n",
      "[6300 / 314403] events analyzed...\n",
      "[6350 / 314403] events analyzed...\n",
      "[6400 / 314403] events analyzed...\n",
      "[6450 / 314403] events analyzed...\n",
      "[6500 / 314403] events analyzed...\n",
      "[6550 / 314403] events analyzed...\n",
      "[6600 / 314403] events analyzed...\n",
      "[6650 / 314403] events analyzed...\n",
      "[6700 / 314403] events analyzed...\n",
      "[6750 / 314403] events analyzed...\n",
      "[6800 / 314403] events analyzed...\n",
      "[6850 / 314403] events analyzed...\n",
      "[6900 / 314403] events analyzed...\n",
      "[6950 / 314403] events analyzed...\n",
      "[7000 / 314403] events analyzed...\n",
      "[7050 / 314403] events analyzed...\n",
      "[7100 / 314403] events analyzed...\n",
      "[7150 / 314403] events analyzed...\n",
      "[7200 / 314403] events analyzed...\n",
      "[7250 / 314403] events analyzed...\n",
      "[7300 / 314403] events analyzed...\n",
      "[7350 / 314403] events analyzed...\n",
      "[7400 / 314403] events analyzed...\n",
      "[7450 / 314403] events analyzed...\n",
      "[7500 / 314403] events analyzed...\n",
      "[7550 / 314403] events analyzed...\n",
      "[7600 / 314403] events analyzed...\n",
      "[7650 / 314403] events analyzed...\n",
      "[7700 / 314403] events analyzed...\n",
      "[7750 / 314403] events analyzed...\n",
      "[7800 / 314403] events analyzed...\n",
      "[7850 / 314403] events analyzed...\n",
      "[7900 / 314403] events analyzed...\n",
      "[7950 / 314403] events analyzed...\n",
      "[8000 / 314403] events analyzed...\n",
      "[8050 / 314403] events analyzed...\n",
      "[8100 / 314403] events analyzed...\n",
      "[8150 / 314403] events analyzed...\n",
      "[8200 / 314403] events analyzed...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[8250 / 314403] events analyzed...\n",
      "[8300 / 314403] events analyzed...\n",
      "[8350 / 314403] events analyzed...\n",
      "[8400 / 314403] events analyzed...\n",
      "[8450 / 314403] events analyzed...\n",
      "[8500 / 314403] events analyzed...\n",
      "[8550 / 314403] events analyzed...\n",
      "[8600 / 314403] events analyzed...\n",
      "[8650 / 314403] events analyzed...\n",
      "[8700 / 314403] events analyzed...\n",
      "[8750 / 314403] events analyzed...\n"
     ]
    }
   ],
   "source": [
    "!python3 differential_splicing.py \\\n",
    "--genome \"/private/groups/brookslab/reference_sequence/GRCh38.p13.genome.fa\" \\\n",
    "--manifest \"/private/groups/brookslab/rausharm/RBPSig/star_output_03.11.24/bam_manifest.txt\" \\\n",
    "--annotations \"/private/groups/brookslab/reference_annotations/gencode.v37.annotation.gtf\" \\\n",
    "--output-prefix \"/private/groups/brookslab/rausharm/RBPSig/03.11.24\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dba67ed3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "52fc90cf",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20340ec2",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
